### PR TITLE
feat: 6770 new column type switch

### LINKED ISSUE #6770: [Feature] : Support switch as a column type in table widget
## Summary

As a dev, I should be able to add the following as column types

- Switch

# Requirements

| Requirement   | User Story | Importance | Notes |
| ------------- | ------------- | ------------- | ------------- |
| Column type: `switch` | Sets the column type to switch, alter default selected | High  | - |
| default value | The default value for switch; Boolean |  High | - |
| Disabled | Should the input remain disabled | high | - |
| onChange | trigger event on change| High | - |
| Alignment | Alignment inside the table cell | Medium | - |

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> Enable New Column type switch in table

Fixes #6770 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Tested locally.
> Added test cases.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: FEATURE/6770-new-column-type-switch 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.14 **(0.01)** | 37.2 **(0.03)** | 34.02 **(0.05)** | 55.64 **(0)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx | 87.06 **(0.16)** | 55.22 **(0.33)** | 79.63 **(0)** | 87.06 **(0.16)**
 :red_circle: | app/client/src/widgets/TableWidget/component/TableUtilities.tsx | 37.57 **(-0.38)** | 24.79 **(0.9)** | 14.29 **(-1.93)** | 35.93 **(-0.25)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/derived.js | 65.74 **(7.09)** | 62.79 **(3.03)** | 50 **(4.76)** | 63.82 **(5.18)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/getTableColumns.tsx | 56 **(25.57)** | 23.08 **(23.08)** | 33.33 **(33.33)** | 50 **(30)**
 :red_circle: | app/client/src/widgets/TableWidget/widget/index.tsx | 29.03 **(-0.85)** | 9.77 **(-0.03)** | 29.09 **(-0.91)** | 28.57 **(-0.97)**
 :red_circle: | app/client/src/widgets/TableWidget/widget/propertyConfig.ts | 90 **(-5.74)** | 100 **(0)** | 86.05 **(-7.28)** | 90 **(-5.74)**
 :green_circle: | app/client/src/widgets/TableWidget/widget/propertyUtils.ts | 25.56 **(1.13)** | 8 **(0)** | 20 **(0)** | 24.19 **(1.24)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/widget/index.tsx [hunk 1]
```diff
@@ -619,6 +719,33 @@ class TableWidget extends BaseWidget<TableWidgetProps, WidgetState> {
       newPrimaryColumns = this.createTablePrimaryColumns();
       if (newPrimaryColumns) this.updateColumnProperties(newPrimaryColumns);
     }
+
+    // set defaultOptionValue for column type "select" if exist
+    const columnIds = Object.keys(this.props.primaryColumns);
+    const editedColumnData = cloneDeep(this.props.editedColumnData);
```
Comment (SatishGandham) [inline app/client/src/widgets/TableWidget/widget/index.tsx:725]: Is cloneDeep required here, as long as the evaluation is triggered immer will take care of this.

**Review by SatishGandham — CHANGES_REQUESTED** [review]
Issue only talks about adding the switch column type.
Are we doing the select as well?

**Comment by vibhandikyash** [comment]
@SatishGandham Actually that is another ticket but since we need some common code to be used we are merging that branch here.

**Review by SatishGandham — DISMISSED** [review]

**Comment by vibhandikyash** [comment]
Closing this PR. Feature will be reopened with TableV2.

**Pull request closed without merging**